wordpress-mobile/WordPress-Android
# Feature/8796 create site ui

Fixes #8796 

**NOTE: `feature/8512-create-site-step` is set as target branch. Replace it with `feature/master-site-creation` when #8795 is merged.**

This PR integrates NewSiteCreationService with the NewSitePreviewFragment + introduces `NewSitePreviewViewModel`

To test:
- My Site -> Switch Site -> Plus sign Icon -> Create a WordPress.com Site -> Select a Segment -> Skip Verticals -> Skip Site Info Screen -> type something 

Verify following + try to rotate your device

- turn on the airplane mode -> pick a domain -> no connection error is displayed ->  turn off the airplane mode -> click on retry -> the flow should continue

- pick a domain ->  when the loading indicator is displayed and you leave the app a progress notification is shown + it's automatically dismissed when you reenter the app

- pick a domain ->  when the loading indicator is displayed and you leave the app a progress notification is shown + if you wait until the site is created, the progress notification is replaced with 
"Site Created" notification + Tapping on the notification should redirect you to the app

- pick a domain -> wait until the site is created -> notice the Preview screen is displayed with an already preloaded site in the webview

- pick a domain -> wait until the preview screen is displayed -> click on the "OK" button -> notice you are redirected to the My Site screen and the newly created site is selected

- edit `CreateSiteUseCase` so it fails -> pick a domain -> generic error is shown

- verify the "Cancel/'X'" button on both error screens terminates the flow


Known issues
1) WebView's content is cleared on rotation #8821
2) When the user presses "OK" button and the SiteModel of the newly created site hasn't been fetched from the server -> the user should be navigated to the SitePickerActivity and a toast message should be displayed. #8822
3) The transition animation is missing #8583 
4) There is no timeout set when the site is being preloaded into the webview -> so if it takes 30seconds before the site is fully loaded, the user will see the loading screen for 30s #8824

![creating](https://user-images.githubusercontent.com/2261188/50230340-bb1cdd00-03ac-11e9-9bc3-00230983232a.jpg)
